c7n/resources/cloudtrail.py

```diff
@@ -119,6 +119,129 @@ def process(self, resources, event=None):
     def __call__(self, r):
         return self.match(r[self.annotation_key])
 
+@CloudTrail.filter_registry.register('log-metric-filter-pattern')
+class LogMetricFilterPattern(Filter):
+
+    """
+    If pattern entered is found then resource will pass if not then resource will fail
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: log-metric-filter-pattern
+            resource: aws.cloudtrail
+            filters:
+            - type: log-metric-filter-pattern
+                required_pattern: "{ ($.eventName = \"ConsoleLogin\") && ($.additionalEventData.MFAUsed != \"Yes\") }"
+    """
+    schema = type_schema('log-metric-filter-pattern',
+                            required_pattern = {'type':'string'},
+                            required= ['required_pattern'])
+
+    schema_alias = False
+    permissions = ('cloudtrail:DescribeTrails', 'cloudtrail:GetTrailStatus', 'cloudtrail:GetEventSelectors',
+                   'cloudwatch:DescribeAlarms', 'logs:DescribeMetricFilters', 'sns:ListSubscriptions')
+    annotation_key = 'c7n:CheckForMultiTrailMfaLogin'
+
+    def process(self, resources,event=None):
+        trails_w_multiregion= []
+        trails_w_activeStatus = []
+        trails_w_IncludeManagementEvents =[]
+        filters_matched =[]
+        log_group_names = []
+        match_alarms_actions =[]
+        grouped_trails = get_trail_groups(self.manager.session_factory, resources)
+        for region, (client, trails) in grouped_trails.items():
+            # Ensure there is at least one multi-region CloudTrail
+            if not grouped_trails:
+                self.log.info('No cloudtrails have been created')
+                return resources
+            else:
```

Comment:
you can collapse a lot of these if/else statements by dedenting the else block if you check the condition and return